### PR TITLE
[docs] Add testing section

### PR DESCRIPTION
**Pull request by jordangarcia**
addresses #128

### CODE REVIEW

**Comment by bhamodi** [comment]
Thanks @jordangarcia! :+1:

File: docs/src/docs/08-testing.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+export default { projectsMap, currentProject, currentProjectId }
+```
+
+### Tests
+
+Given our module we want to test the following:
+
+- Using `actions.setCurrentProjectId()` sets the correct id using the `currentProjectId` getter
+
+- When `Api.fetchProducts` is stubbed with mock data, calling `actions.fetchProjects` properly poulates
```
Comment (balloob) [inline docs/src/docs/08-testing.md]: Typo, poulates

File: docs/src/docs/08-testing.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+import Project from './index'
+
+let mockProjects = [
+  { id: '123-abc', name: 'project 1' },
+  { id: '456-cdf', name: 'project 2' },
+]
+
+describe('modules/Project', () => {
+  afterEach(() => {
+    flux.reset()
```
Comment (balloob) [inline docs/src/docs/08-testing.md]: You keep referring to flux but it is never defined. You mean reactor?

File: docs/src/docs/08-testing.md (hunk 1 above)
Comment (bhamodi) [inline docs/src/docs/08-testing.md]: Fixed in #144.

File: docs/src/docs/08-testing.md (hunk 2 above)
Comment (bhamodi) [inline docs/src/docs/08-testing.md]: Good catch @balloob - `flux` is actually a wrapper we use for NuclearJS at Optimizely. In the case of the example tests, I believe it should be referencing `reactor`, yes. I'll send a PR fixing this. Thanks. :+1:

**Pull request merged**